### PR TITLE
update go modules

### PR DESCRIPTION
**Pull request by ipochi**
Updates the builder image version to 1.18.5 and updates the go modules
to mitigate CVEs. The list of vulnerabilities found in the binaries are
listed in [comment](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/372#issuecomment-1208124322).

Signed-off-by: Imran Pochi <imranpochi@microsoft.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,60 +1,74 @@
...
-	google.golang.org/grpc v1.42.0
-	k8s.io/api v0.20.10
-	k8s.io/apimachinery v0.20.10
-	k8s.io/client-go v0.20.10
-	k8s.io/component-base v0.20.10
-	k8s.io/klog/v2 v2.4.0
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.0
+	golang.org/x/net v0.0.0-20220805013720-a33c5aa5df48
+	google.golang.org/grpc v1.48.0
+	google.golang.org/protobuf v1.28.1
```
Comment (cheftako) [inline go.mod:14]: I realize protobug moved. Need to check that there are no additional steps needed when we switch package.

File: go.mod [hunk 2]
```diff
@@ -1,60 +1,74 @@
...
+	k8s.io/api v0.24.3
+	k8s.io/apimachinery v0.24.3
+	k8s.io/client-go v0.24.3
+	k8s.io/component-base v0.24.3
+	k8s.io/klog/v2 v2.70.1
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.32
```
Comment (tallclair) [inline go.mod:20]: Leave the konnectivity-client at `v0.0.0`, since it's using a replace directive.

**Comment by ipochi** [comment]
test pull-apiserver-network-proxy-make-lint

test pull-apiserver-network-proxy-make-test

**Comment by ipochi** [comment]
@tallclair @cheftako 

Finally the lint job succeeds. moved all the components to go v1.18.5. Please review :)

File: go.mod [hunk 3]
```diff
@@ -1,60 +1,74 @@
...
+	google.golang.org/protobuf v1.28.1
+	k8s.io/api v0.24.3
+	k8s.io/apimachinery v0.24.3
+	k8s.io/client-go v0.24.3
+	k8s.io/component-base v0.24.3
+	k8s.io/klog/v2 v2.70.1
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.0
 )
 
 require (
```
Comment (liggitt) [inline go.mod:23]: this added a *lot* of new dependencies, many of which we explicitly don't want to depend on transitively in Kubernetes:

* github.com/go-openapi/...
* new incompatible version of github.com/imdario/mergo (xref https://github.com/kubernetes/kubernetes/issues/107499)

Also, this only updated the root go.mod file, not the client konnectivity-client/go.mod file, so now this repo has two disparate dependency trees. If the client go.mod file was updated the same way this was, we would be unable to pull new versions into kubernetes.

How were the versions of these updated modules selected?

File: cmd/server/app/server.go [hunk 4]
```diff
@@ -350,7 +354,7 @@ func (p *Proxy) runAdminServer(o *options.ProxyRunOptions, server *server.ProxyS
 		}
 	}
 	adminServer := &http.Server{
-		Addr:           net.JoinHostPort(o.AdminBindAddress, strconv.Itoa(o.AdminPort)),
+		Addr:           fmt.Sprintf("127.0.0.1:%d", o.AdminPort),
```
Comment (tallclair) [inline cmd/server/app/server.go:357]: Why was this changed? Looks like maybe a bad rebase?

Comment (cheftako) [reply cmd/server/app/server.go:357]: +1. However this was merged Sept 2, over 3 months ago.

File: go.mod (hunk 3 above)
Comment (tallclair) [reply go.mod:23]: The two that you mentioned are both transitive dependencies through client-go. I don't know how that specific client-go version was selected, but since this PR sat open for a long time I'm guessing it was the latest version when this PR was opened.

We haven't been keeping the konnectivity-client go.mod in sync with this since we're tracking the module versions from the oldest version of k8s we support in the client. We should maybe consider adding a presubmit to ensure that konnectivity-client doesn't take any dependencies on kubernetes staging repos, but hopefully our PR review process would catch that.

https://github.com/kubernetes/client-go/blob/v0.24.3/go.mod#L26

openapi

Comment (liggitt) [reply go.mod:23]: oops, I misread the go-openapi additions, these are the ones we want to avoid:
```
      "github.com/go-openapi/analysis": "use k8s.io/kube-openapi/pkg/validation/spec",
      "github.com/go-openapi/spec": "use k8s.io/kube-openapi/pkg/validation/spec instead",
      "github.com/go-openapi/strfmt": "use k8s.io/kube-openapi/pkg/validation/strfmt instead",
      "github.com/go-openapi/validate": "use k8s.io/kube-openapi/pkg/validation/validate instead",
```

the github.com/imdario/mergo one is a known dependency but the version selected here is known to be problematic (see the linked issue above)
